passes/memory/memory_dff.cc

```diff
@@ -197,7 +198,8 @@ struct MemoryDffWorker
 				if (sigbit_users_count[bit] > 1)
 					goto skip_ff_after_read_merging;
 
-			if (find_sig_before_dff(sig_data, clk_data, clk_polarity, true) && clk_data != RTLIL::SigSpec(RTLIL::State::Sx) && sig_data == check_q)
+			if (find_sig_before_dff(sig_data, clk_data, clk_polarity, true) && clk_data != RTLIL::SigSpec(RTLIL::State::Sx) &&
+					std::all_of(check_q.begin(), check_q.end(), [&](const SigSpec &cq) {return cq == sig_data; }))
```

Comment:
Wait, this checks that all entries in `check_q` are equal to `sig_data`?

Reply:
Yes, at this point `sig_data` is actually the FF output because of `find_sig_before_dff`'s behaviour

Reply:
Ahh never mind, I see what it's doing. Merge when ready.